snuba/datasets/schema.py

```diff
@@ -1,51 +1,88 @@
-    basic metadata for now. The code to generate the schema comes in a followup PR.
+    basic metadata for now.
     """
 
     TEST_TABLE_PREFIX = "test_"
 
     def __init__(self, local_table_name, dist_table_name, columns):
-        self.__columns = columns
+        self._columns = columns
 
         self.__local_table_name = local_table_name
         self.__dist_table_name = dist_table_name
 
-    def __make_test_table(self, table_name):
+    def _make_test_table(self, table_name):
         return table_name if not settings.TESTING else "%s%s" % (self.TEST_TABLE_PREFIX, table_name)
 
     def get_local_table_name(self):
         """
         This returns the local table name for a distributed environment.
         It is supposed to be used in DDL commands and for maintenance.
         """
-        return self.__make_test_table(self.__local_table_name)
+        return self._make_test_table(self.__local_table_name)
 
     def get_table_name(self):
         """
         This represents the table we interact with to send queries to Clickhouse.
         In distributed mode this will be a distributed table. In local mode it is a local table.
         """
         table_name = self.__local_table_name if local_dataset_mode() else self.__dist_table_name
-        return self.__make_test_table(table_name)
+        return self._make_test_table(table_name)
+
+    def get_local_table_definition(self):
+        raise NotImplementedError
+
+    def get_local_drop_table_statement(self):
+        return "DROP TABLE IF EXISTS %s" % self.get_local_table_name()
 
-    def _get_table_definition(self, name, engine):
+    def get_columns(self):
+        return self._columns
+
+    def get_column_differences(self, expected_columns: Mapping[str, str]) -> List[str]:
+        """
+        Returns a list of differences between the expected_columns and the columns described in the schema.
+        """
+        errors: List[str] = []
+
+        for column_name, column_type in expected_columns.items():
+            if column_name not in self._columns:
+                errors.append("Column '%s' exists in local ClickHouse but not in schema!" % column_name)
+                continue
+
+            expected_type = self._columns[column_name].type.for_schema()
+            if column_type != expected_type:
+                errors.append(
+                    "Column '%s' type differs between local ClickHouse and schema! (expected: %s, is: %s)" % (
+                        column_name,
+                        expected_type,
+                        column_type
+                    )
+                )
+
+        return errors
+
+
+class TableSchema(Schema):
+    def _get_table_definition(self, name: str, engine: str) -> str:
         return """
         CREATE TABLE IF NOT EXISTS %(name)s (%(columns)s) ENGINE = %(engine)s""" % {
-            'columns': self.__columns.for_schema(),
+            'columns': self._columns.for_schema(),
```

Comment:
if you have a get_columns method, you do not need to expose this to the subclasses.